packages/default/scss/fab/_layout.scss

```diff
@@ -77,13 +77,31 @@
 
     .k-fab-item {
         list-style-type: none;
-        margin: 4px 0;
+        margin: map-get( $spacing, 1 ) 0;
         outline: none;
         white-space: nowrap;
+        display: flex;
+        align-items: center;
+        justify-content: flex-end;
+
+        &.k-text-right {
+            flex-direction: row;
+
+            .k-fab-item-text {
+                margin-right: map-get( $spacing, 2 ) ;
+            }
+        }
+        &.k-text-left {
+            flex-direction: row-reverse;
+
+            .k-fab-item-text {
+                margin-left: map-get( $spacing, 2 );
+            }
+        }
     }
 
     .k-fab-item-text {
-        padding: 4px 4px;
+        padding: map-get( $spacing, 1 ) map-get( $spacing, 1 );
         border-width: 1px;
         border-radius: 2px;
         font-size: 10px;
```

Comment:
We could use $font-size-xs instead of hardcoded value

Reply:
I extracted all variables for both text and icon in order for clients to customize them as they want.